app/models/factory_log.rb

```diff
@@ -1,103 +1,39 @@
 # This is log model executed factory.
 class FactoryLog < ActiveRecord::Base
-  # TODO: fix algorithm
-  # TODO: fix name devide to name and traits
   # TODO: fix visualize association relation and selectable counting or not.
-  # TODO: initializer setting parent_id.
 
-  # Return Factory's log ranking about total time it took.
-  # They are identified factory_name + trait_name.
-
-  belongs_to :parent, foreign_key: "parent_id", class_name: "FactoryLog"
-  has_many :children, foreign_key: "parent_id", class_name: "FactoryLog"
   belongs_to :factory
 
-  def self.time_ranking
-    ranking = []
+  # Arguments is expected, hash like this,
+  # name: factory_name
+  # traits: trait_name(String) array
+  # assos: array of hash { name:association_name, traits: trait(string) array, factory_name: factory_name }
+  # I wrote factroy_inspector gem (https://github.com/lastcat/factory_inspector). Please use it.
+  def self.logging(inspected_factory, execution_time)
+    #puts Factory.create_unique_factory(inspected_factory)
+    FactoryLog.create(factory_id: Factory.create_unique_factory(inspected_factory).id, execution_time: execution_time)
+  end
+
+  # Return Factory's log ranking about total time it took.
+  def self.ranking
+    results = []
     FactoryLog.all.each do |log|
-      time = log.time
-      name = log.name
-      traits = log.traits
-      if index = ranking.index { |cd| cd[:name] == (name + traits) }
-        hash_data = ranking[index]
+      time = log.execution_time
+      factory = log.factory
+      if index = results.index { |cd| cd[:factory] == factory }
+        hash_data = results[index]
         hash_data[:total_time] += time
         hash_data[:count] += 1
         hash_data[:average_time] = hash_data[:total_time] / hash_data[:count]
       else
-        ranking.push(
-                      name: (name + traits),
+        results.push(
+                      factory: factory,
```

Comment:
Style/FirstParameterIndentation: Indent the first parameter one step more than the previous line.